pkg/kubelet/server/portforward/websocket.go

```diff
@@ -45,21 +45,25 @@ const (
 
 // options contains details about which streams are required for
 // port forwarding.
-type v4Options struct {
-	ports []uint16
+type V4Options struct {
+	Ports []int32
 }
 
 // newOptions creates a new options from the Request.
-func newV4Options(req *http.Request) (*v4Options, error) {
+func NewV4Options(req *http.Request) (*V4Options, error) {
+	if !wsstream.IsWebSocketRequest(req) {
+		return &V4Options{}, nil
+	}
+
 	portStrings := req.URL.Query()[api.PortHeader]
 	if len(portStrings) == 0 {
-		return nil, fmt.Errorf("%q is required", api.PortHeader)
+		return nil, fmt.Errorf("%q header is required", api.PortHeader)
```

Comment:
So the issue and I can fix it is that it is a header for SPDY and a query parameter for WebSockets. I can have 2 even though they are the same name.